idaes/gas_solid_contactors/unit_models/fixed_bed_0D.py

```diff
@@ -0,0 +1,330 @@
+                        EnergyBalanceType,
+                        UnitModelBlockData,
+                        useDefault)
+from idaes.core.util.config import (is_physical_parameter_block)
+import idaes.logger as idaeslog
+from idaes.core.util import get_solver
+
+__author__ = "Chinedu Okoli, Andrew Lee"
+
+
+# Assumptions:
+# Only Vap and Sol phases, which are explicitly named as such
+# Perfect mixing in Vap phase
+# Static solid phase
+
+# Need to have build-on-demand properties in reactions for this to work
+
+@declare_process_block_class("FixedBed0D")
+class FixedBed0DData(UnitModelBlockData):
+    """
+    0D Fixed Bed Reactor Model Class
+    """
+    CONFIG = ConfigBlock()
+    CONFIG.declare("dynamic", ConfigValue(
+        domain=In([True]),
+        default=True,
+        description="Dynamic model flag - must be True",
+        doc="""Indicates whether this model will be dynamic or not,
+**default** = True. Fixed beds must be dynamic."""))
+    CONFIG.declare("has_holdup", ConfigValue(
+        default=True,
+        domain=In([True]),
+        description="Holdup construction flag - must be True",
+        doc="""Indicates whether holdup terms should be constructed or not.
+**default** - True. Fixed bed reactors must be dynamic, thus this must be
+True."""))
+    CONFIG.declare("energy_balance_type", ConfigValue(
+        default=EnergyBalanceType.useDefault,
+        domain=In(EnergyBalanceType),
+        description="Energy balance construction flag",
+        doc="""Indicates what type of energy balance should be constructed,
+**default** - EnergyBalanceType.useDefault.
+**Valid values:** {
+**EnergyBalanceType.useDefault - refer to property package for default
+balance type
+**EnergyBalanceType.none** - exclude energy balances,
+**EnergyBalanceType.enthalpyTotal** - single enthalpy balance for material,
+**EnergyBalanceType.enthalpyPhase** - enthalpy balances for each phase,
+**EnergyBalanceType.energyTotal** - single energy balance for material,
+**EnergyBalanceType.energyPhase** - energy balances for each phase.}"""))
+    CONFIG.declare("gas_property_package", ConfigValue(
+        default=useDefault,
+        domain=is_physical_parameter_block,
+        description="Property package to use for gas phase",
+        doc="""Property parameter object used to define property calculations
+for the gas phase, **default** - useDefault.
+**Valid values:** {
+**useDefault** - use default package from parent model or flowsheet,
+**PhysicalParameterObject** - a PhysicalParameterBlock object.}"""))
+    CONFIG.declare("gas_property_package_args", ConfigBlock(
+        implicit=True,
+        description="Arguments to use for constructing gas phase "
+        "property packages",
+        doc="""A ConfigBlock with arguments to be passed to a gas phase
+property block(s) and used when constructing these,
+**default** - None.
+**Valid values:** {
+see property package for documentation.}"""))
```

Comment:
Update to `ConfigValue` with `default={}`, and `domain=dict`. Take a look at the equivalent config argument in the MB for `property_package_args` as an example.